pkg/trace/api/evp_intake_proxy.go

```diff
@@ -0,0 +1,233 @@
+		msg := fmt.Sprintf("EvpIntakeProxy is disabled: %v", message)
+		http.Error(w, msg, http.StatusMethodNotAllowed)
+	})
+}
+
+// evpIntakeReverseProxyHandler creates an http.ReverseProxy which can forward payloads
+// to one or more endpoints, based on the request received and the Agent configuration.
+// Headers are not proxied, instead we add our own known set of headers.
+// See also evpIntakeProxyTransport below.
+func evpIntakeReverseProxyHandler(conf *config.AgentConfig, endpoints []config.Endpoint, transport http.RoundTripper, logger *stdlog.Logger) http.Handler {
+	director := func(req *http.Request) {
+
+		containerID := req.Header.Get("Datadog-Container-ID")
+		contentType := req.Header.Get("Content-Type")
+		userAgent := req.Header.Get("User-Agent")
+
+		// Clear all received headers
+		req.Header = http.Header{}
+
+		// Standard headers
+		req.Header.Set("Via", fmt.Sprintf("trace-agent %s", info.Version))
+		if contentType != "" {
+			req.Header.Set("Content-Type", contentType)
+		}
+		req.Header.Set("User-Agent", userAgent) // Set even if an empty string so Go doesn't set its default
+		req.Header["X-Forwarded-For"] = nil     // Prevent setting X-Forwarded-For
+
+		// Datadog headers
+		if ctags := getContainerTags(conf.ContainerTags, containerID); ctags != "" {
+			req.Header.Set("X-Datadog-Container-Tags", ctags)
+		}
+		req.Header.Set("X-Datadog-Hostname", conf.Hostname)
+		req.Header.Set("X-Datadog-AgentDefaultEnv", conf.DefaultEnv)
+
+		// URL, Host and the API key header are set in the transport for each outbound request
+	}
+
+	return &httputil.ReverseProxy{
+		Director:  director,
+		ErrorLog:  logger,
+		Transport: &evpIntakeProxyTransport{transport, endpoints},
+	}
+}
+
+// evpIntakeProxyTransport sends HTTPS requests to multiple targets using an
+// underlying http.RoundTripper. API keys are set separately for each target.
+// When multiple endpoints are in use the response from the first endpoint
+// is proxied back to the client, while for all aditional endpoints the
+// response is discarded.
+type evpIntakeProxyTransport struct {
+	transport http.RoundTripper
+	endpoints []config.Endpoint
+}
+
+func (t *evpIntakeProxyTransport) RoundTrip(req *http.Request) (rresp *http.Response, rerr error) {
+
+	// Metrics with stats for debugging
+	beginTime := time.Now()
+	metricTags := []string{}
+	defer func() {
+		metrics.Count("datadog.trace_agent.evpiproxy.request", 1, metricTags, 1)
+		metrics.Count("datadog.trace_agent.evpiproxy.request_bytes", req.ContentLength, metricTags, 1)
+		metrics.Timing("datadog.trace_agent.evpiproxy.request_duration_ms", time.Since(beginTime), metricTags, 1)
+		if rerr != nil {
+			metrics.Count("datadog.trace_agent.evpiproxy.request_error", 1, metricTags, 1)
+		}
+	}()
+
+	// Parse request path: The first component is the target subdomain, the rest is the target path.
+	inputPath := req.URL.Path
+	urlComponents := strings.SplitN(inputPath, "/", 3)
```

Comment:
```suggestion
	parts := strings.SplitN(inputPath, "/", 3)
```
We generally call this `parts`